### ISSUE TITLE
Getting a relationship is inefficient

### ISSUE THREAD

**Issue by poechsel** [tozti/tozti#19]
If someone wants to get a relation, he must proceed this way:
- First, get the id of the relationship by sending a `GET` request to `api/store/resources/<uid>/<rel>`
- Then, get the data corresponding by the id by sending a `GET` request to `api/store/resources/<uid>`
This example is for a `to-one` relationship and already proves to be a bit cumbersome.

Now, imagine the relation is `to-many`. In that case, we must proceed like this:
- First, get a list of id's contained by the relationship by sending a `GET` request to `api/store/resources/<uid>/<rel>`
- Then, as we do not have any way to do batch requests, for each id send a `GET` request to get the content.

This will become inefficient very quickly.

As such, I propose:
- to modify `get` on relationship so that it sends back the object (or the list of object) instead of just the id
- to add a batch `get` for objects.

**Comment by dancsi** [comment]
In principle I agree, as we will never be inspecting a relationship without actually accessing the referenced entities. Probably it would be better to add additional optional parameters to the get request, to go up to n levels deep in the hierarchy when rendering the reply. This will keep the API more or less consistent between all http verbs

**Comment by Lapin0t** [comment]
Yeah, so jsonapi has a story for [that](http://jsonapi.org/format/#fetching-includes). The idea is that to get a thread `X` _and all it's posts_, the request would be `GET /api/store/resources/X?include=posts`. The answer should contain the thread in the `data` property as usual but additionaly the `included` property is a dictionary mapping resource ids to the resource object. It will contain every resource that is a target of the `posts` relationship.

**Comment by Lapin0t** [comment]
Also, something related: i just noticed that in a [relationship object](http://jsonapi.org/format/#document-resource-object-relationships), the api must only include _at least one_ of the `links` (we call it `self`) and `data`, so we could actually write the api so that for _to-many_ relationships we only provide a link (`/api/store/resources/{id}/{rel}`) and not the array of target ids. This would solve the problem of "i want to load a thread but i don't want the list of posts (ids)".

**Comment by liautaud** [comment]
(cc. @jetSett)
